app/src/main/java/com/whereuat/whereu/ContactRetriever.java

```diff
@@ -0,0 +1,69 @@
+package com.whereuat.whereu;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.ContactsContract;
+import android.content.Context;
+
+/**
+ * Created by whites5 on 3/22/16.
+ */
+public class ContactRetriever {
+    private static Context mContext;
+    private static String phone_num;
+    private String contact_id;
+    private String contact_name;
+    private static final String[] projection = {
+        ContactsContract.Contacts._ID,
+        ContactsContract.Contacts.DISPLAY_NAME
+    };
+
+
+    public ContactRetriever(int reqCode, int resultCode, Intent data, Context con) {
+        mContext = con;
+        switch (reqCode) {
+            case (1) :
+                if (resultCode == Activity.RESULT_OK) {
+                    Uri contactData = data.getData();
+
+                    Cursor c =  con.getContentResolver().query(contactData, projection, null, null, null);
+                    if (c.moveToFirst()) {
+                        contact_id = c.getString(c.getColumnIndex(ContactsContract.Contacts._ID));
+                        contact_name = c.getString(c.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME));
+                    }
+                    c.close();
+
+                    Cursor cursorPhone = con.getContentResolver().query(ContactsContract.CommonDataKinds.Phone.CONTENT_URI,
+                            new String[]{ContactsContract.CommonDataKinds.Phone.NUMBER},
+
+                            ContactsContract.CommonDataKinds.Phone.CONTACT_ID + " = ? AND " +
```

Comment:
This is kind of hard to read. I'd make a new variable for this and then just pass that string in
```
String contact_query = String.format("%s = ? AND %s = %s", ContactsContract.CommonDataKinds.Phone.CONTACT_ID, ContactsContract.CommonDataKinds.Phone.TYPE, ContactsContract.CommonDataKinds.Phone.TYPE_MOBILE);
```
Also, maybe consider changing your imports so you import `ContactsContract.CommonDataKinds` because it's really obnoxious having to qualify these names every time.